### ISSUE TITLE
cannot create MV over simple agg MV

### ISSUE THREAD

**Issue by skyzh** [singularity-data/risingwave#4067]
**Describe the bug**

```
create table t1(v1 int);
create materialized view mv1 as select sum(v1) as sum_v1 from t1;
create materialized view mv2 as select * from mv1;
```

**Expected behavior**

**Additional context**

**Comment by BowenXiao1999** [comment]
Invesitigating. The pk_columns of mv1 is empty. Not sure is it expected? 
```sql
                                   QUERY PLAN                                   
--------------------------------------------------------------------------------
 StreamMaterialize { columns: [sum(count)(hidden), sum_v1], pk_columns: [] }
   StreamGlobalSimpleAgg { aggs: [sum($0), sum($1)] }
     StreamExchange { dist: Single }
       StreamLocalSimpleAgg { aggs: [count, sum($0)] }
         StreamTableScan { table: t1, columns: [v1, _row_id], pk_indices: [1] }
```

**Comment by BowenXiao1999** [comment]
Investigating. Some info:

I guess the error comes from here: 
https://github.com/singularity-data/risingwave/blob/9d8524d159935355051c8d57437ca6d7526ec93c/src/meta/src/stream/scheduler.rs#L235-L241

Let's say the actor 9 of singleton MV1 is running on parallel unit 0. Here we generate 4 actors for chain, running on parallel unit 1, 2, 3, 4. 

Therefore, when proceed to resolve ChainNode, the parallel unit (1, 2, 3, 4) is not able to found upstream actor id 9. (Because the only mapping is parallel unit 0 -> actor 9).

I think we should enable `same_worker_node_as_upstream` = true in this case, but for some reason we failed. And we may not need 4 actors to scan 1 value... So there should be something wrong for the placement and schedule.

**Comment by BugenZhao** [comment]
Oops, sorry for missing the assignment. Investigation from @BowenXiao1999 really makes sense. I've drafted a fix for this in #4153.
